src/main/java/com/direwolf20/buildinggadgets/common/building/RegionSpliterator.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    private RegionSpliterator(int minX, int minY, int minZ, int maxX, int maxY, int maxZ) {
+        this(minX, minY, minZ, maxX, maxY, maxZ, minX, minY, minZ, true);
+    }
+
+    private RegionSpliterator(int minX, int minY, int minZ, int maxX, int maxY, int maxZ, int posX, int posY, int posZ, boolean allowYZSplit) {
+        this.minX = minX;
+        this.minY = minY;
+        this.minZ = minZ;
+        this.maxX = maxX;
+        this.maxY = maxY;
+        this.maxZ = maxZ;
+
+        this.nextPosX = posX;
+        this.nextPosY = posY;
+        this.nextPosZ = posZ;
+        this.allowYZSplit = allowYZSplit;
+    }
+
+    @Override
+    public boolean tryAdvance(Consumer<? super BlockPos> action) {
+        if (isXOverflowed()) {
+            return false;
+        }
+
+        this.allowYZSplit = false;
+        BlockPos pos = new BlockPos(nextPosX, nextPosY, nextPosZ);
+
+        nextPosZ++;
+        if (isZOverflowed()) {
+            nextPosZ = minZ;
+            nextPosY++;
+        } else {
+            action.accept(pos);
+            return true;
+        }
+
+        if (isYOverflowed()) {
+            nextPosY = minY;
+            nextPosX++;
+        } else {
+            action.accept(pos);
+            return true;
+        }
+
+        //(maxX, maxY, maxZ)
+        action.accept(pos);
+        return true;
+    }
+
+    /**
+     * @return A part of this {@link Spliterator}, if splitting is possible
+     */
+    @Override
+    public Spliterator<BlockPos> trySplit() {
+        int oldMinX = minX;
+        int oldMinY = minY;
+        int oldMinZ = minZ;
+        int oldPosX = nextPosX;
+        int oldPosY = nextPosY;
+        int oldPosZ = nextPosZ;
+
+        //Construct new min coordinates, so that at least one can be split of: max - min >= 1
+        if (maxX > minX) {
+            //As Region's coordinates are inclusive, the amount of blocks along one axis is max - min + 1
+            //Half the length + base x
+            minX = (maxX - minX + 1) / 2 + minX + 1;
+            resetPos();
+            return new RegionSpliterator(oldMinX, oldMinY, oldMinZ, minX - 1, maxY, maxZ, oldPosX, oldPosY, oldPosZ, allowYZSplit);
+        } else if (maxY > minY && allowYZSplit) {
+            minY = (maxY - minY + 1) / 2 + minY + 1;
+            resetPos();
+            return new RegionSpliterator(oldMinX, oldMinY, oldMinZ, maxX, minY - 1, maxZ, oldPosX, oldPosY, oldPosZ, allowYZSplit);
+        } else if (maxZ > minZ && allowYZSplit) {
+            minZ = (maxZ - minZ + 1) / 2 + minZ + 1;
+            resetPos();
+            return new RegionSpliterator(oldMinX, oldMinY, oldMinZ, maxX, maxY, minZ - 1, oldPosX, oldPosY, oldPosZ, allowYZSplit);
+        }
+        return null;
+    }
+
+    @Override
+    public long estimateSize() {
+        return (long) (Math.abs(maxX - nextPosX) + 1) * (long) (Math.abs(maxY - nextPosY) + 1) * (long) (Math.abs(maxZ - nextPosZ + 1));
```

Comment:
Note that `|a + b| + 1 != |a + b + 1|`, for example `|-4 + -2| + 1 = 7`, and `|-4 + -2 + 1| = 5`.
What is possible is create a absolute method for long's.

Reply:
Also since `minX`, `minY`, `minZ`, `maxX`, `maxY`, and `maxZ` are int's, their sum/difference wouldn't overflow (as badly).
I don't think somebody will try to create a region closing the whole world, and a typical minecraft world is far smaller than 2^32-1.

Reply:
I missed the +1, sorry.